torch/_subclasses/meta_utils.py

```diff
@@ -441,6 +450,7 @@ def empty_create(inner_t, inner_src):
                                     AttrSource(source, attr),
                                 )
                             ),
+                            extra_context=extra_context,
```

Comment:
I don't understand what's going on here, ISTR we talked about this a little but I've forgotten, can you remind me?

Reply:
The problem is that:

* `transform_subclass()` is supposed to take a wrapper subclass, and produce a new wrapper subclass
* It also takes in a lambda, telling it how each inner tensor should be transformed.
* This isn't enough though - `transform_subclass` assumes that the **only** thing that changes from the input wrapper subclass to the output wrapper subclass is the values of the inner tensors (e.g. real -> fake). But for JaggedTensor, the extra `SymInt` corresponding to the jagged size needs to come from somewhere, and doesn't exist on the input jagged subclass (well... I think it does, but it doesn't have a proper `Source`?)


I also want to confirm my understanding: the jagged size is an actual `SymNode` in **both** eager and in torch.compile, but we can't just re-use the input `JaggedTensor`'s `SymNode` when we fakeify, because the eager mode jagged size doesn't have a source attached to it?

Reply:
> I also want to confirm my understanding: the jagged size is an actual SymNode in both eager and in torch.compile, but we can't just re-use the input JaggedTensor's SymNode when we fakeify, because the eager mode jagged size doesn't have a source attached to it? Since sources are specific to dynamo.

Hmm it depends on what you by "actual SymNode". Jagged tensor's ragged size is a SymInt holding a SymNode, but not actually the SymNode defined in Python! There's a pybinded SymNode class previously only used for XLA that we're coopting (see `torch/csrc/jit/python/init.cpp`). That is also why when we fakify we need to create a real python "class SymNode" using the pybinded SymNode as val.